src/com/unbounce/treajure/concurrent/mvar.clj

```diff
@@ -0,0 +1,82 @@
+(ns com.unbounce.treajure.concurrent.mvar
+  "Haskell's MVar abstraction in Clojure.
+
+  Check: http://chimera.labs.oreilly.com/books/1230000000929/ch07.html for more
+  info around MVars."
+  (:import [java.util.concurrent Semaphore]))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Lightweight Performance mutable tuple implementation in Clojure
+;; (Look ma! no atoms!)
+
+(defprotocol IMCons
+  (mcar [p])
+  (mcdr [p])
+  (set-mcar! [p val])
+  (set-mcdr! [p val]))
+
+(deftype MCons [^{:volatile-mutable true} car
+                ^{:volatile-mutable true} cdr]
```

Comment:
Uh Lisp :crying_cat_face: 

OK, cool, you may want to drop the pull request then?